itest/loadtest/mint_batch_test.go

```diff
@@ -166,3 +169,198 @@ func mintTest(t *testing.T, ctx context.Context, cfg *Config) {
 
 	itest.SyncUniverses(ctx, t, bob, alice, aliceHost, cfg.TestTimeout)
 }
+
+// mintTestV2 checks that we can mint a batch of assets. It is a more
+// performant version of the existing mintTest, as it uses less assertions and
+// RPC calls.
+func mintTestV2(t *testing.T, ctx context.Context, cfg *Config) {
+	// Start by initializing all our client connections.
+	alice, bob, bitcoinClient := initClients(t, ctx, cfg)
+
+	// We query the assets of each node once on this step. Every function
+	// that needs to take a node's assets into account will be passed these
+	// values instead of calling the RPC again. This is done to minimize
+	// collateral RPC impact of the loadtest.
+	resAlice, err := alice.ListAssets(ctx, &taprpc.ListAssetRequest{})
+	require.NoError(t, err)
+
+	resBob, err := bob.ListAssets(ctx, &taprpc.ListAssetRequest{})
+	require.NoError(t, err)
+
+	assetsAlice := resAlice.Assets
+	assetsBob := resBob.Assets
+
+	totalAssets := make([]*taprpc.Asset, len(assetsAlice)+len(assetsBob))
+	copy(totalAssets, assetsAlice)
+	copy(totalAssets[len(assetsAlice):], assetsBob)
+
+	// Alice serves as the minter.
+	minter := alice
+
+	// First we make sure group initialization is completed. We check if
+	// there's any more groups left
+	existingGroups := getTotalAssetGroups(totalAssets)
+	groupKeys := make(map[string][]byte, 0)
+
+	for _, v := range existingGroups {
+		tweakedKey, err := hex.DecodeString(v)
+		require.NoError(t, err)
+
+		groupKeys[v] = tweakedKey
+	}
+
+	var remainingGroups int
+	if cfg.TotalNumGroups > len(existingGroups) {
+		remainingGroups = cfg.TotalNumGroups - len(existingGroups)
+	}
+
+	t.Logf("Existing groups=%v, minting %v new groups",
+		len(existingGroups), remainingGroups)
+	for range remainingGroups {
+		mintNewGroup(t, ctx, bitcoinClient, minter, cfg)
+	}
+
+	// If there's not any existing groups we skip the rest of the steps, we
+	// will mint into those groups in another run.
+	if len(existingGroups) == 0 {
+		return
+	}
+
+	groupIndex := rand.Intn(len(existingGroups))
+	groupKey := groupKeys[existingGroups[groupIndex]]
+
+	mintIntoGroup(t, ctx, bitcoinClient, minter, groupKey, cfg)
+}
+
+// mintNewGroup mints an asset that creates a new group.
+func mintNewGroup(t *testing.T, ctx context.Context, miner *rpcclient.Client,
+	minter *rpcClient, cfg *Config) []*taprpc.Asset {
+
+	mintAmt := rand.Uint64() % uint64(cfg.MintSupplyMax)
+	if mintAmt < uint64(cfg.MintSupplyMin) {
+		mintAmt = uint64(cfg.MintSupplyMin)
+	}
+
+	// nolint:lll
+	assetRequests := []*mintrpc.MintAssetRequest{
```

Comment:
nit: can remove a level of indentation by formatting as:
```
	assetRequests := []*mintrpc.MintAssetRequest{{
		Asset: &mintrpc.MintAsset{
			AssetType: taprpc.AssetType_NORMAL,
			Name: fmt.Sprintf(
				"tapcoin-%d", time.Now().UnixNano(),
			),
			AssetMeta: &taprpc.AssetMeta{
				Data: []byte("{}"),
				Type: taprpc.AssetMetaType_META_TYPE_JSON,
			},
			Amount:          mintAmt,
			NewGroupedAsset: true,
			DecimalDisplay:  4,
		},
	}}
```
That way we don't need the `lll` nolint directive. Same below.